Extend Graph, CompilationUnit, and schema matching to accept optional operator version number

tugsbayasgalan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #70396
* #68358
* #69915
* __->__ #69914

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D33198157](https://our.internmc.facebook.com/intern/diff/D33198157)

tugsbayasgalan committed:
Extend Graph, CompilationUnit, and schema matching to accept optional operator version number

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Extend Graph, CompilationUnit, and schema matching to accept optional operator version number"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Extend Graph, CompilationUnit, and schema matching to accept optional operator version number"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33198157).

tugsbayasgalan committed:
Update on "Extend Graph, CompilationUnit, and schema matching to accept optional operator version number"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D33198157](https://our.internmc.facebook.com/intern/diff/D33198157)

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33198157).

tugsbayasgalan committed:
Update on "Extend Graph, CompilationUnit, and schema matching to accept optional operator version number"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D33198157](https://our.internmc.facebook.com/intern/diff/D33198157)

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33198157).

tugsbayasgalan committed:
Update on "Extend Graph, CompilationUnit, and schema matching to accept optional operator version number"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D33198157](https://our.internmc.facebook.com/intern/diff/D33198157)

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33198157).

tugsbayasgalan committed:
Update on "Extend Graph, CompilationUnit, and schema matching to accept optional operator version number"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D33198157](https://our.internmc.facebook.com/intern/diff/D33198157)

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33198157).

qihqi commented on torch/csrc/jit/frontend/schema_matching.cpp:
```diff
@@ -616,12 +631,48 @@ Value* emitBuiltinCall(
+        }
+      }
+    }
+    if (!found_upgrader)
```
wrap with {}

gmagogsfm commented on torch/csrc/jit/frontend/schema_matching.cpp:
```diff
@@ -616,12 +631,48 @@ Value* emitBuiltinCall(
+          upgrader_schemas.push_back(old_schema);
+          found_upgrader = true;
+        } else {
+          if (!isOpCurrentBasedOnUpgraderEntries(
+                  version_entry->second, graph_version.value())) {
+            TORCH_INTERNAL_ASSERT(false, "Valid upgrader must be present");
```
Is this also about deprecated upgrader? If so, the error message can be improved. Current error message doesn't explain what happened and isn't actionable.

Besides, `torch_internal_assert` implies that something is wrong with TorchScript implementation (ie some invariant that should never be broken is broken), can we throw a normal exception here instead?

tugsbayasgalan committed:
Update on "Extend Graph, CompilationUnit, and schema matching to accept optional operator version number"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D33198157](https://our.internmc.facebook.com/intern/diff/D33198157)

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33198157).

tugsbayasgalan committed:
Update on "Extend Graph, CompilationUnit, and schema matching to accept optional operator version number"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D33198157](https://our.internmc.facebook.com/intern/diff/D33198157)

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33198157).

tugsbayasgalan committed:
Update on "Extend Graph, CompilationUnit, and schema matching to accept optional operator version number"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D33198157](https://our.internmc.facebook.com/intern/diff/D33198157)

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33198157).

tugsbayasgalan committed:
Update on "Extend Graph, CompilationUnit, and schema matching to accept optional operator version number"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D33198157](https://our.internmc.facebook.com/intern/diff/D33198157)

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33198157).
